cmds/boot/systemboot/main.go

```diff
@@ -208,14 +242,65 @@ func addSEL(sequence string) {
 	}
 }
 
+// getDebugEnabled checks whether debug output is requested, either via command line or via VPD
+// variables.
+// If -q was explicitly passed on the command line, will use that value, otherwise will look for
+// the VPD variable "systemboot_log_level".
+// Valid values are coreboot loglevels https://review.coreboot.org/cgit/coreboot.git/tree/src/commonlib/include/commonlib/loglevel.h,
+// either as integer (1, 2) or string ("debug").
+// If the VPD variable is missing or it is set to an invalid value, it will use the default.
+func getDebugEnabled() bool {
+	if isFlagPassed("q") {
+		return !*doQuiet
+	}
+
+	// -q was not passet, so `doQuiet` contains the default value
+	defaultDebugEnabled := !*doQuiet
+	// check for the VPD variable "systemboot_log_level". First the read-write, then the read-only
+	v, err := vpd.Get(vpdSystembootLogLevel, false)
+	if err != nil {
+		// TODO do not print warning if file is not found
+		log.Printf("Warning: failed to read read-write VPD variable '%s', will try the read-only one. Error was: %v", vpdSystembootLogLevel, err)
+		v, err = vpd.Get(vpdSystembootLogLevel, true)
+		if err != nil {
+			// TODO do not print warning if file is not found
+			log.Printf("Warning: failed to read read-only VPD variable '%s', will use the default value. Error was: %v", vpdSystembootLogLevel, err)
+			return defaultDebugEnabled
+		}
+	}
+	level := strings.ToLower(strings.TrimSpace(string(v)))
+	// allow integer and translate to string level
```

Comment:
good idea but i'd move it to the "default" branch of the switch statement below

Reply:
not sure I follow, then I need to do the mapping of levels to debug on/off again?

Reply:
strconv stuff should be attempted only if the string didn't match any of the known levels, i.e. in the default branch of the switch statement below.

Reply:
oh, sorry, i misread. the logic is backwards.
ok, ignore this.